### PR TITLE
Add net/http integration to trace the outgoing requests

### PR DESCRIPTION
**Pull request by rakyll**

### CODE REVIEW

File: plugins/http/httptrace/httptrace.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	req = req.WithContext(trace.WithSpan(req.Context(), span))
+	resp, err := t.base().RoundTrip(req)
+	// TODO(jbd): Add status and attributes.
+	span.End()
+	return resp, err
+}
+
+// CancelRequest cancels an in-flight request by closing its connection.
+func (t *Transport) CancelRequest(req *http.Request) {
+	type canceler interface {
```
Comment (odeke-em) [inline plugins/http/httptrace/httptrace.go]: Could we hoist this out of this method?

Comment (rakyll) [reply plugins/http/httptrace/httptrace.go]: Not going to optimize anything. It is self-contained here.

Comment (odeke-em) [reply plugins/http/httptrace/httptrace.go]: Aye aye, cool :)

**Comment by rakyll** [comment]
/cc @jba

We will also use this plugin in the tracing of the google-api-go-client clients.

**Comment by jba** [comment]
LGTM.

File: plugins/http/httptrace/httptrace.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+// Transport is an http.RoundTripper that traces the outgoing requests.
+type Transport struct {
+	// Base is the base http.RoundTripper to be used to do the actual request.
+	//
+	// Optional. If nil, http.DefaultTransport is used.
+	Base http.RoundTripper
+}
+
+// RoundTrip creates a trace.Span and inserts it into the outgoing request's headers.
```
Comment (nvartolomei) [inline plugins/http/httptrace/httptrace.go]: isn't it missing `inserts it into the outgoing request's headers` part?

Comment (rakyll) [reply plugins/http/httptrace/httptrace.go]: True, we probably shouldn't merge this until we sort out how to determine which vendor specific header we should inject depending on the registered injectors. I am going to label it as a "do not review".

**Comment by rakyll** [comment]
Closed via https://github.com/census-instrumentation/opencensus-go/commit/6c0bc0d00b45f8e5e7a4cfc70071722739f1f1b1.

**Pull request closed without merging**